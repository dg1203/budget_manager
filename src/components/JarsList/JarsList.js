import React from "react";
import PropTypes from "prop-types";
import { Grid, Card, CardContent, Typography, Button } from "@material-ui/core";
import { Link } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  margin: {
    marginBottom: "16px"
  }
});

const JarsList = ({ jars }) => {
  const classes = useStyles();
  return (
    <Grid container spacing={3}>
      {jars.map(jar => (
        <Grid key={jar.id} item xs={12} sm={6} md={4}>
          <Card>
            <CardContent>
              <Typography
                className={classes.margin}
                variant="h5"
                component="h2"
              >
                {jar.name}
              </Typography>
              <Typography className={classes.margin} color="textSecondary">
                {jar.description}
              </Typography>
              <Typography
                className={classes.margin}
                variant="body2"
                component="p"
              >
                Elementów w słoiku: {jar.resources.length}
                <br />
                {jar.currency !== "" ? `Waluta: ${jar.currency}` : ""}
              </Typography>
              <Link className={classes.margin} to={`/jar/edit/${jar.id}`}>
                <Button variant="contained" color="primary" fullWidth>
                  Więcej
                </Button>
              </Link>
            </CardContent>
          </Card>
        </Grid>
      ))}
    </Grid>
  );
};

JarsList.propTypes = {
  jars: PropTypes.array.isRequired
};

export default JarsList;
