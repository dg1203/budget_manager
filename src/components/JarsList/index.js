import JarsList from "./JarsList";

export default JarsList;