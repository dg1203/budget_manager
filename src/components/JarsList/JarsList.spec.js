import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import JarsList from "./JarsList";
import store from "../../store";

const defaultProps = {
  jars: [
    {
      currency: "PLN",
      description: "Jar 1",
      id: "3xzty3489",
      isDefault: false,
      name: "Name 2",
      resources: []
    },
    {
      currency: "",
      description: "Jar 2",
      id: "3xsty3489",
      isDefault: false,
      name: "Name 2",
      resources: [
        {
          PLN: 250
        }
      ]
    }
  ]
};

const renderComponent = () =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <JarsList {...defaultProps} />
      </MemoryRouter>
    </Provider>
  );

describe("JarsList component", () => {
  it("renders the component", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
