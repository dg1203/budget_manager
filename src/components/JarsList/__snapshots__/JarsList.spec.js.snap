// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JarsList component renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root makeStyles-margin-1 MuiTypography-h5"
              >
                Name 2
              </h2>
              <p
                class="MuiTypography-root makeStyles-margin-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
              >
                Jar 1
              </p>
              <p
                class="MuiTypography-root makeStyles-margin-1 MuiTypography-body2"
              >
                Elementów w słoiku: 
                0
                <br />
                Waluta: PLN
              </p>
              <a
                class="makeStyles-margin-1"
                href="/jar/edit/3xzty3489"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Więcej
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <h2
                class="MuiTypography-root makeStyles-margin-1 MuiTypography-h5"
              >
                Name 2
              </h2>
              <p
                class="MuiTypography-root makeStyles-margin-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
              >
                Jar 2
              </p>
              <p
                class="MuiTypography-root makeStyles-margin-1 MuiTypography-body2"
              >
                Elementów w słoiku: 
                1
                <br />
                
              </p>
              <a
                class="makeStyles-margin-1"
                href="/jar/edit/3xsty3489"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Więcej
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root makeStyles-margin-1 MuiTypography-h5"
            >
              Name 2
            </h2>
            <p
              class="MuiTypography-root makeStyles-margin-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Jar 1
            </p>
            <p
              class="MuiTypography-root makeStyles-margin-1 MuiTypography-body2"
            >
              Elementów w słoiku: 
              0
              <br />
              Waluta: PLN
            </p>
            <a
              class="makeStyles-margin-1"
              href="/jar/edit/3xzty3489"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Więcej
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root makeStyles-margin-1 MuiTypography-h5"
            >
              Name 2
            </h2>
            <p
              class="MuiTypography-root makeStyles-margin-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Jar 2
            </p>
            <p
              class="MuiTypography-root makeStyles-margin-1 MuiTypography-body2"
            >
              Elementów w słoiku: 
              1
              <br />
              
            </p>
            <a
              class="makeStyles-margin-1"
              href="/jar/edit/3xsty3489"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Więcej
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
